### PR TITLE
Fix deprecated code for TokenProvider

### LINKED ISSUE #479: Fix Jwts usage claims without deprecated method
##### **Overview of the issue**

Fix Jwts usage claims without deprecated method in TokenProvider

##### **Motivation for or Use Case**

Code without deprecated method

### PR DESCRIPTION
**Pull request by abdelmoghit1**
Delete parser deprecated method and use parserBuilder

Fix #479

- Please make sure the below checklist is followed for Pull Requests.

- [ ] [Travis tests](https://travis-ci.org/jhipster/jhipster-registry/pull_requests) are green
- [ ] Tests are added where necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/jhipster-registry/blob/main/CONTRIBUTING.md) are followed

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
     "@typescript-eslint/parser": "2.11.0",
     "autoprefixer": "9.7.3",
     "base-href-webpack-plugin": "2.0.0",
-    "browser-sync": "2.26.7",
+    "browser-sync": "^2.26.12",
```
Comment (pascalgrimaud) [inline package.json:47]: use fixed version

File: package.json [hunk 2]
```diff
@@ -55,14 +55,14 @@
     "eslint-loader": "3.0.3",
     "file-loader": "5.0.2",
     "friendly-errors-webpack-plugin": "1.7.0",
-    "generator-jhipster": "6.8.0",
+    "generator-jhipster": "^6.10.3",
```
Comment (pascalgrimaud) [inline package.json:58]: you should not upgrade this, as all the generated code is not updated

File: package.json [hunk 3]
```diff
@@ -55,14 +55,14 @@
...
     "html-loader": "0.5.5",
     "html-webpack-plugin": "3.2.0",
     "husky": "3.1.0",
     "jest": "24.9.0",
     "jest-date-mock": "1.0.7",
     "jest-junit": "10.0.0",
-    "jest-preset-angular": "8.0.0",
+    "jest-preset-angular": "^8.3.1",
```
Comment (pascalgrimaud) [inline package.json:65]: use fixed version

File: package.json [hunk 4]
```diff
@@ -85,7 +85,7 @@
     "webpack": "4.41.2",
     "webpack-bundle-analyzer": "3.6.0",
     "webpack-cli": "3.3.10",
-    "webpack-dev-server": "3.9.0",
+    "webpack-dev-server": "^3.11.0",
```
Comment (pascalgrimaud) [inline package.json:88]: use fixed version

File: src/main/java/io/github/jhipster/registry/security/jwt/TokenProvider.java [hunk 5]
```diff
@@ -97,7 +94,7 @@ public Authentication getAuthentication(String token) {
 
     public boolean validateToken(String authToken) {
         try {
-            Jwts.parser().setSigningKey(key).parseClaimsJws(authToken);
+        	Jwts.parserBuilder().setSigningKey(key).build().parseClaimsJws(authToken);
```
Comment (pascalgrimaud) [inline src/main/java/io/github/jhipster/registry/security/jwt/TokenProvider.java:97]: indent

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

**Comment by pascalgrimaud** [comment]
Thanks for your contribution @abdelmoghit1
I did some minor comment. If you can fix it, so I can do a new release of JHipster Registry with your changes

**Review by pascalgrimaud — APPROVED** [review]

File: src/main/java/io/github/jhipster/registry/security/jwt/TokenProvider.java (hunk 5 above)
Comment (abdelmoghit1) [reply src/main/java/io/github/jhipster/registry/security/jwt/TokenProvider.java]: will the PR be merged automatically? i see you approved the changes but the branch is not yet merged

Comment (pascalgrimaud) [reply src/main/java/io/github/jhipster/registry/security/jwt/TokenProvider.java]: it's good :)

**Pull request merged**